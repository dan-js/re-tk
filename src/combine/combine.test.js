import React from "react";
import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import { combine } from ".";

describe("combine()", () => {
    it("renders the correct tree", () => {
        const Combined = combine([
            ["div", { "data-testid": "c1" }],
            ["article", { "data-testid": "c2" }],
            ["span", { "data-testid": "c3" }],
        ]);

        const { baseElement } = render(<Combined>innermost!</Combined>);

        expect(screen.getByTestId("c1")).toBeInTheDocument();
        expect(screen.getByTestId("c2")).toBeInTheDocument();
        expect(screen.getByTestId("c3")).toBeInTheDocument();

        expect(screen.getByTestId("c3")).toHaveTextContent("innermost!");

        expect(baseElement.innerHTML).toMatchSnapshot();
    });

    it("renders the correct tree with functional components", () => {
        // eslint-disable-next-line react/prop-types
        const Wrap = ({ testid, children }) => {
            return <div data-testid={testid}>{children}</div>;
        };

        const Combined = combine([
            [Wrap, { testid: "c1" }],
            [Wrap, { testid: "c2" }],
            [Wrap, { testid: "c3" }],
        ]);

        const { baseElement } = render(<Combined>innermost!</Combined>);

        expect(screen.getByTestId("c1")).toBeInTheDocument();
        expect(screen.getByTestId("c2")).toBeInTheDocument();
        expect(screen.getByTestId("c3")).toBeInTheDocument();

        expect(screen.getByTestId("c3")).toHaveTextContent("innermost!");

        expect(baseElement.innerHTML).toMatchSnapshot();
    });
});
