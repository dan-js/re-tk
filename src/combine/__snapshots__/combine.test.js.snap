// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`combine() renders the correct tree 1`] = `"<div><div data-testid=\\"c1\\"><article data-testid=\\"c2\\"><span data-testid=\\"c3\\">innermost!</span></article></div></div>"`;

exports[`combine() renders the correct tree with functional components 1`] = `"<div><div data-testid=\\"c1\\"><div data-testid=\\"c2\\"><div data-testid=\\"c3\\">innermost!</div></div></div></div>"`;
